src/main/java/com/openlattice/chronicle/services/upload/AppDataUploadManagerImpl.java

```diff
@@ -0,0 +1,654 @@
+        if ( organizationId != null ) {
+            return scheduledTasksManager.userAppsFullNamesByOrg.getOrDefault( packageName, ImmutableSet.of() )
+                    .contains( organizationId );
+        }
+        return scheduledTasksManager.userAppsFullNameValues.contains( packageName );
+    }
+
+    // create entities and edges
+    private void createEntitiesAndAssociations(
+            DataApi dataApi,
+            DataIntegrationApi dataIntegrationApi,
+            List<SetMultimap<UUID, Object>> data,
+            UUID organizationId,
+            UUID studyId,
+            UUID deviceEKID,
+            String participantId,
+            String deviceId,
+            UUID participantEKID,
+            UUID participantESID
+    ) {
+        // entity set ids
+        UUID appDataESID = commonTasksManager
+                .getEntitySetId( organizationId, CHRONICLE_DATA_COLLECTION, APPDATA, DATA_ES );
+        UUID recordedByESID = commonTasksManager
+                .getEntitySetId( organizationId, CHRONICLE_DATA_COLLECTION, RECORDED_BY, RECORDED_BY_ES );
+        UUID devicesESID = commonTasksManager
+                .getEntitySetId( organizationId, CHRONICLE_DATA_COLLECTION, DEVICE, DEVICES_ES );
+        UUID usedByESID = commonTasksManager
+                .getEntitySetId( organizationId, CHRONICLE_DATA_COLLECTION, USED_BY, USED_BY_ES );
+        UUID userAppsESID = commonTasksManager
+                .getEntitySetId( organizationId, CHRONICLE_DATA_COLLECTION, USER_APPS, USER_APPS_ES );
+
+        ListMultimap<UUID, Map<UUID, Set<Object>>> appDataEntities = ArrayListMultimap.create();
+        ListMultimap<UUID, DataAssociation> appDataAssociations = ArrayListMultimap.create();
+
+        Map<EntityKey, Map<UUID, Set<Object>>> entitiesByEntityKey = Maps.newHashMap();
+        Set<Triple<EntityKey, EntityKey, EntityKey>> edgesByEntityKey = Sets.newHashSet();
+
+        OffsetDateTime timeStamp = OffsetDateTime.now();
+
+        for ( int i = 0; i < data.size(); i++ ) {
+            SetMultimap<UUID, Object> appEntity = data.get( i );
+            appDataEntities.put( appDataESID, Multimaps.asMap( appEntity ) );
+
+            appDataAssociations
+                    .putAll( getAppDataAssociations( deviceEKID,
+                            participantEKID,
+                            appDataESID,
+                            devicesESID,
+                            recordedByESID,
+                            participantESID,
+                            timeStamp,
+                            i ) );
+
+            String appPackageName, appName;
+            appPackageName = appName = getFirstValueOrNullByUUID( appEntity, FULL_NAME_FQN );
+            String dateLogged = getMidnightDateTime( getFirstValueOrNullByUUID( appEntity, DATE_LOGGED_FQN ) );
+
+            if ( scheduledTasksManager.systemAppPackageNames.contains( appPackageName ) || dateLogged == null )
```

Comment:
we should always use braces around if statements as a general style thing, but especially on a 2 line if statement / when using keywords like this